engine/framework/model/master.go

```diff
@@ -52,7 +53,10 @@ type MasterMetaKVData struct {
 
 	// Config holds business-specific data
 	Config []byte `json:"config" gorm:"column:config;type:blob"`
-	// TODO: add master status and checkpoint data
+
+	// if job is failed, `ExtMsg` can hold the error message
```

Comment:
Seems this is a requirement for business logic, can the framework do it?

Reply:
It's the business logic.
What I mean is `it can hold`, but depends to business logic.

Reply:
I will make it more understandable.